### PR TITLE
Converted install guides chapter 3 to mod docs

### PR DESCRIPTION
**Pull request by tahliar**
Bug 1825224 - Modularise "Performing Additional Configuration on Satellite Server"

https://bugzilla.redhat.com/show_bug.cgi?id=1825224

### CODE REVIEW

**Comment by tahliar** [comment]
Looks like Travis's broken links today are just projects.theforeman.org, which is expected. I changed a couple of links that I thought would be reported as broken until this is published... the mysteries of Travis :shrug:

File: guides/common/assembly_using-external-databases.adoc [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+ifeval::["{build}" == "satellite"]
+Red Hat does not provide support or tools for external database maintenance. This includes backups, upgrades, and database tuning. You must have your own database administrator to support and maintain external databases.
+endif::[]
+
+// Known issue: need to explicitly include the section titles so that two xrefs in the same paragraph render correctly
+Use xref:mongodb-as-an-external-database-considerations_{context}[MongoDB as an External Database Considerations] and xref:postgresql-as-an-external-database-considerations_{context}[PostgreSQL as an External Database Considerations] to decide if you want to use external databases for your {Project} deployment.
+
+[id="external-databases-installation-overview_{context}"]
+== Installation Overview
```
Comment (spetrosi) [inline guides/common/assembly_using-external-databases.adoc:18]: Lines 17 and 18 seems redundant. Also, FCC does not allow to use two IDs within one assembly. Removing those two lines will solve it.

File: guides/common/modules/con_mongodb-as-an-external-database-considerations.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
+[id="mongodb-as-an-external-database-considerations_{context}"]
+= MongoDB as an External Database Considerations
+
+Pulp uses the MongoDB database. If you want to use MongoDB as an external database, the following information can help you discern if this option is right for your {Project} configuration. {Project} supports MongoDB version 3.4.
```
Comment (spetrosi) [inline guides/common/modules/con_mongodb-as-an-external-database-considerations.adoc:4]:
```suggestion
Pulp uses the MongoDB database. If you want to use MongoDB as an external database, the following information can help you decide if this option is right for your {Project} configuration. {Project} supports MongoDB version 3.4.
```

For simplicity

File: guides/common/modules/con_postgresql-as-an-external-database-considerations.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[id="postgresql-as-an-external-database-considerations_{context}"]
+= PostgreSQL as an External Database Considerations
+
+Foreman, Katello, and Candlepin use the PostgreSQL database. If you want to use PostgreSQL as an external database, the following information can help you discern if this option is right for your {Project} configuration. {Project} supports PostgreSQL version 9.2.
```
Comment (spetrosi) [inline guides/common/modules/con_postgresql-as-an-external-database-considerations.adoc:4]:
```suggestion
Foreman, Katello, and Candlepin use the PostgreSQL database. If you want to use PostgreSQL as an external database, the following information can help you decide if this option is right for your {Project} configuration. {Project} supports PostgreSQL version 9.2.
```

File: guides/common/modules/performing_initial_configuration_sat_server_manual_disconnected.adoc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 [[performing_initial_configuration_sat_server_manual_disconnected]]
 = Performing the Initial Configuration Manually
 
-This initial configuration procedure creates an organization, location, user name, and password. After the initial configuration, you can create additional organizations and locations if required. The initial configuration also installs MongoDB and PostgreSQL databases on the same server. Depending on your deployment, using external databases can benefit performance. For more information see, xref:using-external-databases[].
+This initial configuration procedure creates an organization, location, user name, and password. After the initial configuration, you can create additional organizations and locations if required. The initial configuration also installs MongoDB and PostgreSQL databases on the same server. Depending on your deployment, using external databases can benefit performance. For more information see, link:{BaseURL}installing_satellite_server_from_a_connected_network/performing_additional_configuration_on_satellite_server#using-external-databases_satellite[Using External Databases with {Project}].
```
Comment (spetrosi) [inline guides/common/modules/performing_initial_configuration_sat_server_manual_disconnected.adoc:4]:
```suggestion
This initial configuration procedure creates an organization, location, user name, and password. After the initial configuration, you can create additional organizations and locations if required. The initial configuration also installs MongoDB and PostgreSQL databases on the same server. Depending on your deployment, using external databases can benefit performance. For more information see, {BaseURL}installing_satellite_server_from_a_connected_network/performing_additional_configuration_on_satellite_server#using-external-databases_satellite[Using External Databases with {Project}].
```
Why did you add `link:` here? This is not common throughout our docs.

File: guides/common/modules/proc_configuring-dns-dhcp-and-tftp.adoc [hunk 5]
```diff
@@ -81,10 +81,4 @@ ifeval::["{context}" == "{smart-proxy-context}"]
...
 
 
-For more information about configuring DHCP, DNS, and TFTP services, see the link:https://access.redhat.com/documentation/en-us/red_hat_satellite/{ProductVersion}/html/provisioning_guide/configuring_networking#Configuring_Networking-Configuring_Network_Services_for_PXE_Boot[Configuring Network Services] section in the _Provisioning Guide_.
-
-ifeval::["{context}" == "{project-context}"]
-For more information about disabling the DNS, DHCP, and TFTP services in {Project} and managing them manually, see xref:disabling_dns_dhcp_tftp_for_unmanaged_networks[Disabling DNS, DHCP, and TFTP for Unmanaged Networks].
-
-For more information about configuring external services, see xref:configuring-external-services[].
-endif::[]
+For more information about configuring DHCP, DNS, and TFTP services, see the link:{BaseURL}provisioning_guide/configuring_networking#Configuring_Networking-Configuring_Network_Services_for_PXE_Boot[Configuring Network Services] section in the _Provisioning Guide_.
```
Comment (spetrosi) [inline guides/common/modules/proc_configuring-dns-dhcp-and-tftp.adoc:84]:
```suggestion
For more information about configuring DHCP, DNS, and TFTP services, see the {BaseURL}provisioning_guide/configuring_networking#Configuring_Networking-Configuring_Network_Services_for_PXE_Boot[Configuring Network Services] section in the _Provisioning Guide_.
```

File: guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc [hunk 6]
```diff
@@ -5,7 +5,7 @@ In a disconnected environment, you must ensure that {ProjectServer} contains the
 
 .Procedure
 
-. Log on to the Red{nbsp}Hat Customer Portal at https://access.redhat.com.
+. Log in to the Red{nbsp}Hat Customer Portal at https://access.redhat.com.
```
Comment (spetrosi) [inline guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]:
```suggestion
. Log on to the Red{nbsp}Hat Customer Portal at https://access.redhat.com.
```
IBM Style Guide p. 342:
log in to v
Write as shown - three words. Use only in documentation for UNIX systems, such as AIX. Otherwise, use
“log on to.”

File: guides/common/modules/proc_importing-kickstart-repositories.adoc [hunk 7]
```diff
@@ -5,7 +5,7 @@ Kickstart repositories are not provided by the Content ISO image. To use Kicksta
 
 .Procedure
 
-. Navigate to the Red{nbsp}Hat Customer Portal at https://access.redhat.com/ and log in.
+. Log in to the Red Hat Customer Portal at https://access.redhat.com.
```
Comment (spetrosi) [inline guides/common/modules/proc_importing-kickstart-repositories.adoc:8]:
```suggestion
. Log on to the Red Hat Customer Portal at https://access.redhat.com.
```

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+. Use the instructions in link:{BaseURL}installing_satellite_server_from_a_connected_network/#attaching-satellite-infrastructure-subscription_satellite[Attaching the {Project} Infrastructure Subscription] to attach a {Project} subscription to your server.
+
+. Disable all repositories and enable only the following repositories:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# subscription-manager repos --disable '*'
+# subscription-manager repos --enable=rhel-server-rhscl-7-rpms \
+--enable=rhel-7-server-rpms
```
Comment (spetrosi) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:22]: Please use attributes for those two repositories

File: guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc (hunk 6 above)
Comment (tahliar) [reply guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]: We use "log in to" in many other places (found more of them than I found "log on to"), and I think it's appropriate since we do write Linux docs.

Comment (spetrosi) [reply guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]: As @melcorr explained, you log in to a linux system, but log on on some website.

File: guides/common/assembly_using-external-databases.adoc (hunk 1 above)
Comment (tahliar) [reply guides/common/assembly_using-external-databases.adoc:18]: This is a sub-chapter heading, since the overview isn't a standalone module but does need a heading. It's the same thing as the heading of a module but is just in the text rather than included.

**Comment by ekohl** [comment]
Unrelated note, but I want to share https://chris.beams.io/posts/git-commit/ and in particular point 5: Use the imperative mood in the subject line. As documentation writers I'm sure you'll appreciate that the past tense triggers me a bit :) Feel free to ignore this, but I'm trying to get the entire Foreman project to agree on this.

File: guides/common/modules/performing_initial_configuration_sat_server_manual_disconnected.adoc (hunk 4 above)
Comment (tahliar) [reply guides/common/modules/performing_initial_configuration_sat_server_manual_disconnected.adoc:4]: Doesn't make a difference in the rendered output but does add highlighting to help me see these {BaseURL} links in my text editor :)

**Comment by melcorr** [comment]
> Unrelated note, but I want to share https://chris.beams.io/posts/git-commit/ and in particular point 5: Use the imperative mood in the subject line. As documentation writers I'm sure you'll appreciate that the past tense triggers me a bit :) Feel free to ignore this, but I'm trying to get the entire Foreman project to agree on this.

This article was always the standard we used with Satellite docs, and the first thing @swadeley prescribed but with time, standards slip :)

File: guides/common/modules/performing_initial_configuration_sat_server_manual_disconnected.adoc (hunk 4 above)
Comment (spetrosi) [reply guides/common/modules/performing_initial_configuration_sat_server_manual_disconnected.adoc:4]: Understood, please keep it then.

File: guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc (hunk 6 above)
Comment (tahliar) [reply guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]: The buttons on the Customer Portal itself say "Log in".

File: guides/common/modules/proc_importing-kickstart-repositories.adoc (hunk 7 above)
Comment (melcorr) [reply guides/common/modules/proc_importing-kickstart-repositories.adoc:8]: Docs always say log on, unless referring to the specific UI element? This would break with wider consistency.

File: guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc (hunk 6 above)
Comment (melcorr) [reply guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]: Docs always say log on, unless referring to the specific UI element? This would break with wider consistency.

Comment (tahliar) [reply guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]: Would it? 

site:access.redhat.com "log in to the Customer Portal" 
4,050 results

site:access.redhat.com "log on to the Customer Portal"
165 results

Comment (melcorr) [reply guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]: Lol!! Does 4050 wrongs make a right though? I'm not sure that it does. It's an easy sed also.

Comment (tahliar) [reply guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]: I mean yeah I definitely think the button on the Customer Portal saying "Log in" is the more compelling of my two arguments haha

Comment (melcorr) [reply guides/common/modules/proc_configuring-satellite-to-synchronize-content-with-a-local-cdn-server.adoc:8]: I'm less inclined to quote IBM style since I have doffed my TW cap, but I do recall a rule that we do not copy over errors from UI elements into the docs. I would argue that the "in" is an error. I would argue that UX really need to start following conventions, because they don't seem to. I would argue that we have a rule and we're not describing the button but the action, and then I would climb back off my soap box and leave you both alone.

File: guides/common/assembly_using-external-databases.adoc (hunk 1 above)
Comment (spetrosi) [reply guides/common/assembly_using-external-databases.adoc:18]: The same kind of overview is given in 3.6. Configuring Foreman server with a Custom SSL Certificate, but this time without a special sub-title for it. I think it is fine for it to have no subtitle because the title of the assembly itself is descriptive: 3.7. Using External Databases with Foreman. 
Now when I look at this Installation Overview, I think we can simple remove it, because modules go in the order that users must perform them. A similar list was required in 3.6. Configuring Foreman server with a Custom SSL Certificate because its last step is to complete procedures in another guide. What do you think?

Comment (tahliar) [reply guides/common/assembly_using-external-databases.adoc:18]: I'd rather keep in the overview steps, just because not all of our current assemblies require using all the procedures in order so I'd like to be super clear, but I agree that we can remove the heading. :+1:

Comment (tahliar) [reply guides/common/assembly_using-external-databases.adoc]: Added a commit to remove the heading. Looks pretty nice! Let me know if you're happy with this or still want to remove the list.
